privilege/privileges/privileges_test.go

```diff
@@ -2464,3 +2464,11 @@ func TestPlacementPolicyStmt(t *testing.T) {
 	mustExec(t, se, dropStmt)
 
 }
+
+func TestDBNameCaseSensitivityInTableLevel(t *testing.T) {
+	store, clean := newStore(t)
+	defer clean()
+	se := newSession(t, store, dbName)
+	mustExec(t, se, "CREATE USER test_user")
+	mustExec(t, se, "grant select on metrics_schema.up to test_user;")
```

Comment:
Test whether it works.

Reply:
It works, but an old test case thinks dbname should be case sensitive, I've modified the case, PTAL.

```go
	// The database name should also case sensitive match.
	_, err = tk.Exec("grant Select,Insert on Test.xx to 'genius'")
	c.Assert(terror.ErrorEqual(err, infoschema.ErrTableNotExists), IsTrue)
```

Reply:
That's weird. I tried on MySQL and it's case insensitive. @tiancaiamao I saw it was your code, do you know why?

Reply:
I believe the table name should be case sensitive.

```
mysql> grant select on PERFORMANCE_SCHEMA.threads to 'xxx';
ERROR 1146 (42S02): Table 'PERFORMANCE_SCHEMA.threads' doesn't exist

mysql> grant select on performance_schema.threads to 'xxx';
Query OK, 0 rows affected (0.01 sec)
```

So this is the expected behaviour:

```
mysql> grant select on METRICS_SCHEMA to xxx;
ERROR 1146 (42S02): Table 'information_schema.METRICS_SCHEMA' doesn't exist


mysql> grant select on METRICS_SCHEMA.up to xxx;
Query OK, 0 rows affected (0.01 sec)
```

Reply:
I think it depends on `lower_case_table_names` and store engine. ref: https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html

1.  lower_case_table_names
mysql default value is `lower_case_table_names=0` on linux (different from mac&windows), which is case sensitive 
when table or db created by `CREATE TABLE or CREATE DATABASE` statement (not system db&table)

tidb default value is `lower_case_table_names=2` on linux, which means name comparisons are not case-sensitive, so we can't create database `test` and `TEST` both.

2.  store engine
`PERFORMANCE_SCHEMA.threads` used a special  engine named ` ENGINE=PERFORMANCE_SCHEMA`, if you use `grant select on information_schema.tables to 'XXX'` things are different. 
For this case `METRICS_SCHEMA.up` is a normal table with a normal engine, so case `PERFORMANCE_SCHEMA.threads` is not suitable I think.

Given all that， I think it's more like a bug, if we regard `METRICS_SCHEMA.up` as a normal table as others.

Reply:
By the way, since tidb only support `lower_case_table_names=2` (ref: https://docs.pingcap.com/zh/tidb/stable/mysql-compatibility), Maybe we should make table name comparison not case-sensitive too? @djshow832 @tiancaiamao

Reply:
> tidb default value is lower_case_table_names=2 on linux

OK, this change looks good to me.
But be careful: the behaviour change may break the compatibility of the old versioned tidb.
If we fix this bug and break a lot of customer's production code, it's problematic.

>  Maybe we should make table name comparison not case-sensitive too

It involves a lot aspects, not a simple change for now.

Reply:
PTAL @djshow832

Reply:
Sorry, I prefer to close it for now, since it's tested by myself and no user requires it. We can reopen and discuss it when it's really a problem.